src/Aspire.Dashboard/Otlp/Model/OtlpApplication.cs

```diff
@@ -68,13 +70,17 @@ public void AddMetrics(AddContext context, RepeatedField<ScopeMetrics> scopeMetr
                             Context = Context
                         };
 
-                        instrument.AddMetrics(metric, ref tempAttributes);
+                        instrument = instrumentRef;
                     }
                     catch (Exception ex)
                     {
-                        context.FailureCount++;
-                        Context.Logger.LogInformation(ex, "Error adding metric.");
+                        // If we can't create the instrument then all data points for it are failures.
+                        context.FailureCount += GetMetricDataPointCount(metric);
+                        Context.Logger.LogInformation(ex, "Error adding metric instrument {MetricName}.", metric.Name);
+                        return;
```

Comment:
Is it OK to skip the metrics coming after this invalid one? Or should this be:

```suggestion
                        continue;
```